File: recursive_addition/csharp_recursive_addition/recursive_addition.cs
using System;

public class RecursiveAdd {
  public static void Main( string[] args ) {

    if( args.Length != 0 ) {
      Console.Error.WriteLine("Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>");
      Environment.Exit(1);
    }

    //Console.Write(args[0] + "+" + args[1] + "=");
    //int x = int.Parse(args[0]);
    //int y = int.Parse(args[1]);
    int x = 6;
    int y = 8;
    Console.WriteLine("rec_add(" + x + "," + y + ")=" + rec_add(x, y));
  }

  private static int rec_add( int x, int y ) {
    if( y == 0 ) {
      return x;
    } else {
      return (1 + rec_add(x, y - 1));
    }
  }
}

File: recursive_addition/fsharp_recursive_addition/recursive_addition.fs
#light
open System

//comment
(* commment *)

let rec rec_add x  y =
    if y = 0 then
      x
    else
      1 + (rec_add x (y - 1))

let rec fact x =
    if x < 1 then 1
    else x * fact (x - 1)

[<EntryPoint>]
let main args =
  let argv = Environment.GetCommandLineArgs()

  if argv.Length <> 1 then
    eprintfn "Usage: %s <noargs>" argv.[0]
    -1
  else
    printfn "rec_add(6,8)=%A" (rec_add 6 8)
    0